doc/ref/files.xml

```diff
@@ -341,8 +341,11 @@ true
 <Description>
 <Ref Func="IsDirectoryPath"/>
 returns <K>true</K> if the file with the filename <A>filename</A> exists
-<E>and</E> is a directory,
-and <K>false</K> otherwise.
+<E>and</E> is a directory, <K>false</K> if it exists and is not a file,
+and <K>fail</K> otherwise. If <Ref Func="IsDirectoryPath"/> returns <K>fail</K>,
+you can use <Ref Func="LastSystemError"/> to get more information about the
+nature of the failure.
```

Comment:
I am quite concerned that this will break code that works right now. In particular, I expect this to be a typical code pattern which would be broken by this change:
```gap
  if not IsDirectoryPath(path) then
    CreatDir(path);
  fi;
```
Now, of course that's bad code to start with (it should have taken into account the possibility that the path exists but isn't a directory). But it can work, and people might not appreciate the breakage.

So wouldn't it be better to have a new function, perhaps a wrapper around `stat`, which takes a path and returns information about it -- such as whether the path refers to a file, a directory, a symlink, .... etc., or simply nothing?